tests/test_routes.py

```diff
@@ -177,3 +178,106 @@ def test_get_wishlist_list(self):
+        )
+        self.assertEqual(
+            new_wishlist_item["product_name"],
+            wishlist_item.product_name,
+            "Product Name does not match",
+        )
+        self.assertEqual(
+            new_wishlist_item["product_price"],
+            str(wishlist_item.product_price),
+            "Product Price does not match",
+        )
+        self.assertEqual(
+            new_wishlist_item["quantity"],
+            wishlist_item.quantity,
+            "Quantity does not match",
+        )
+        self.assertEqual(
+            datetime.strptime(new_wishlist_item["created_date"], '%a, %d %b %Y %H:%M:%S GMT').date(),
+            wishlist_item.created_date,
+            "Created Date does not match",
+        )
+        
+
+    def test_create_wishlist_item_bad_request(self):
+        """It should not create a Wishlist Item when sending the wrong data"""
+        wishlist = WishlistFactory()
+        resp_wishlist = self.client.post(BASE_URL, json=wishlist.serialize())
+        self.assertEqual(resp_wishlist.status_code, status.HTTP_201_CREATED)
+        wishlist_data = resp_wishlist.get_json()
+
+        # Attempt to create a Wishlist Item with incomplete data
+        resp = self.client.post(
+            f'{BASE_URL}/{wishlist_data["id"]}/items',
+            json={"item_property": "Example Property"},  # Incomplete data
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_create_wishlist_item_not_found(self):
+        """It should not create a Wishlist Item for a Wishlist that is not found"""
+        resp = self.client.post(
+            f"{BASE_URL}/0/items",
+            json={"item_property": "Example Property"},
+            content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_create_wishlist_item_unsupported_media_type(self):
+        """It should not create a Wishlist Item when sending the wrong media type"""
+        wishlist = WishlistFactory()
+        resp_wishlist = self.client.post(BASE_URL, json=wishlist.serialize())
+        self.assertEqual(resp_wishlist.status_code, status.HTTP_201_CREATED)
+        wishlist_data = resp_wishlist.get_json()
+
+        # Attempt to create a Wishlist Item with the wrong media type
+        resp = self.client.post(
+            f'{BASE_URL}/{wishlist_data["id"]}/items',
+            json={"item_property": "Example Property"},
+            content_type="test/html"  # Incorrect media type
+        )
+        self.assertEqual(resp.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+        self.assertIsNotNone(resp.get_json())
+
+    def test_create_wishlist_item_method_not_allowed(self):
+        """It should not allow an illegal method call for creating Wishlist Item"""
+        resp = self.client.get(f'{BASE_URL}/0/items')
+        self.assertEqual(resp.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
```

Comment:
I don't think this is a great test? because "get" will be valid for this path to list all items under a specific wishlist.
Correct me if I'm wrong though.

Reply:
Agreed @studentgrad0419.  In fact this test should return a 404 since we are relying on no wishlist with id == 0 in our 404 test (test_create_wishlist_item_not_found).

Reply:
You're right, it's not really a valid test. I removed it.